receiver/httpreceiver/logs.go

```diff
@@ -0,0 +1,232 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpreceiver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+	"unicode"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+)
+
+type httpLogsReceiver struct {
+	path              string
+	serverSettings    *confighttp.HTTPServerSettings
+	telemetrySettings component.TelemetrySettings
+	server            *http.Server
+	consumer          consumer.Logs
+	wg                *sync.WaitGroup
```

Comment:
Usually wait groups don't need to be pointers until they are passed into functions. Can we change this to a non-pointer? That also means you don' need to initialize it.